File: PYTHON/Lee717.py
from typing import List


class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        if bits[-1] != 0:
            return False

        bits.pop()
        pos = 0
        while pos != len(bits):
            if bits[pos] == 1:
                bits.pop(pos)
                if pos == len(bits):
                    return False
                bits.pop(pos)
            else:
                pos += 1
        return True

File: CPP/Lee717.cpp
//
// Created by yangf on 2022/2/20.
//
#include <vector>
#include <list>
using namespace std;
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        if (bits.back() != 0) {
            return false;
        }
        bits.pop_back();
        auto iter = bits.begin();

        while (iter != bits.end()) {
            if (*iter == 1) {
                if (iter == bits.end() - 1) {
                    return false;
                }
                bits.erase(iter, iter + 2);
            } else {
                iter++;
            }
        }
        return true;
    }
};
